{"kind":"symbol","abstract":[{"text":"Called when an error occurs during TrustlessSDK operations.","type":"text"}],"sections":[],"metadata":{"role":"symbol","symbolKind":"method","modules":[{"name":"TrustlessSDK"}],"externalID":"s:12TrustlessSDK0A11SDKDelegateP17didEncounterErroryyAA0aF0OF","title":"didEncounterError(_:)","fragments":[{"text":"func","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"didEncounterError"},{"text":"(","kind":"text"},{"preciseIdentifier":"s:12TrustlessSDK0A5ErrorO","text":"TrustlessError","kind":"typeIdentifier"},{"kind":"text","text":")"}],"required":true,"roleHeading":"Instance Method"},"hierarchy":{"paths":[["doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK","doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/TrustlessSDKDelegate"]]},"primaryContentSections":[{"declarations":[{"tokens":[{"text":"func","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"didEncounterError"},{"kind":"text","text":"("},{"kind":"externalParam","text":"_"},{"kind":"text","text":" "},{"kind":"internalParam","text":"error"},{"text":": ","kind":"text"},{"text":"TrustlessError","kind":"typeIdentifier","identifier":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/TrustlessError","preciseIdentifier":"s:12TrustlessSDK0A5ErrorO"},{"kind":"text","text":")"}],"languages":["swift"],"platforms":["iOS"]}],"kind":"declarations"},{"parameters":[{"content":[{"type":"paragraph","inlineContent":[{"type":"text","text":"A "},{"type":"codeVoice","code":"TrustlessError"},{"type":"text","text":" indicating the type of error that occurred. Implement this method"},{"type":"text","text":" "},{"type":"text","text":"to handle errors specific to TrustlessSDK operations, such as displaying error messages to the user or"},{"text":" ","type":"text"},{"text":"attempting to recover from the error. This method is optional; if not implemented, no action is taken by default.","type":"text"}]}],"name":"error"}],"kind":"parameters"},{"kind":"content","content":[{"level":2,"anchor":"discussion","text":"Discussion","type":"heading"},{"name":"Note","type":"aside","style":"note","content":[{"type":"paragraph","inlineContent":[{"type":"text","text":""},{"type":"codeVoice","code":"TrustlessError"},{"text":" is an enumeration that should define various error states that TrustlessSDK","type":"text"},{"text":" ","type":"text"},{"text":"might encounter (e.g., network errors, authentication failures).","type":"text"}]}]}]}],"variants":[{"paths":["\/documentation\/trustlesssdk\/trustlesssdkdelegate\/didencountererror(_:)"],"traits":[{"interfaceLanguage":"swift"}]}],"identifier":{"interfaceLanguage":"swift","url":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/TrustlessSDKDelegate\/didEncounterError(_:)"},"schemaVersion":{"major":0,"patch":0,"minor":3},"references":{"doc://com.staq.TrustlessSDK/documentation/TrustlessSDK/TrustlessError":{"url":"\/documentation\/trustlesssdk\/trustlesserror","kind":"symbol","abstract":[{"type":"text","text":"Describes errors that can occur within the TrustlessSDK."}],"navigatorTitle":[{"text":"TrustlessError","kind":"identifier"}],"title":"TrustlessError","role":"symbol","identifier":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/TrustlessError","type":"topic","fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"TrustlessError"}]},"doc://com.staq.TrustlessSDK/documentation/TrustlessSDK/TrustlessSDKDelegate":{"url":"\/documentation\/trustlesssdk\/trustlesssdkdelegate","fragments":[{"kind":"keyword","text":"protocol"},{"kind":"text","text":" "},{"text":"TrustlessSDKDelegate","kind":"identifier"}],"identifier":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/TrustlessSDKDelegate","title":"TrustlessSDKDelegate","kind":"symbol","abstract":[{"type":"codeVoice","code":"TrustlessSDKDelegate"},{"text":" protocol defines the delegate methods that ","type":"text"},{"code":"TrustlessSDK","type":"codeVoice"},{"text":" uses to communicate","type":"text"},{"type":"text","text":" "},{"text":"with its delegate object. Implement these methods to handle session expiration and errors specifically","type":"text"},{"text":" ","type":"text"},{"type":"text","text":"related to the TrustlessSDK operations."}],"type":"topic","navigatorTitle":[{"kind":"identifier","text":"TrustlessSDKDelegate"}],"role":"symbol"},"doc://com.staq.TrustlessSDK/documentation/TrustlessSDK":{"type":"topic","kind":"symbol","title":"TrustlessSDK","url":"\/documentation\/trustlesssdk","role":"collection","abstract":[],"identifier":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK"},"doc://com.staq.TrustlessSDK/documentation/TrustlessSDK/TrustlessSDKDelegate/didEncounterError(_:)":{"url":"\/documentation\/trustlesssdk\/trustlesssdkdelegate\/didencountererror(_:)","fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"didEncounterError"},{"kind":"text","text":"("},{"text":"TrustlessError","kind":"typeIdentifier","preciseIdentifier":"s:12TrustlessSDK0A5ErrorO"},{"text":")","kind":"text"}],"type":"topic","abstract":[{"type":"text","text":"Called when an error occurs during TrustlessSDK operations."}],"title":"didEncounterError(_:)","role":"symbol","required":true,"kind":"symbol","identifier":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/TrustlessSDKDelegate\/didEncounterError(_:)"}}}